src/gluonts/model/san/_estimator.py

```diff
@@ -0,0 +1,214 @@
+from typing import List, Optional
+
+import numpy as np
+from mxnet.gluon import HybridBlock
+from gluonts.core.component import validated
+from gluonts.dataset.common import DataEntry
+from gluonts.dataset.field_names import FieldName
+from gluonts.model.estimator import GluonEstimator
+from gluonts.model.predictor import RepresentableBlockPredictor
+from gluonts.support.util import copy_parameters
+from gluonts.time_feature import (
+    TimeFeature,
+    get_lags_for_frequency,
+    time_features_from_frequency_str,
+)
+from gluonts.transform import (
+    AddAgeFeature,
+    AddObservedValuesIndicator,
+    AddTimeFeatures,
+    AsNumpyArray,
+    Chain,
+    ExpectedNumInstanceSampler,
+    InstanceSplitter,
+    RemoveFields,
+    SetField,
+    Transformation,
+    VstackFeatures,
+    ExpandDimArray,
+)
+
+from ._network import (
+    SelfAttentionTrainingNetwork,
+    SelfAttentionPredictionNetwork,
+)
+from ._engine import (
+    Trainer,
+    QuantileForecastGenerator,
+)
+
+
+class SelfAttentionEstimator(GluonEstimator):
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        prediction_length: int,
+        cardinalities: Optional[List[int]] = None,
+        context_length: Optional[int] = None,
+        trainer: Trainer = Trainer(),
+        model_dim: int = 64,
+        ffn_dim_multiplier: int = 2,
+        num_heads: int = 4,
+        num_layers: int = 3,
+        num_outputs: int = 3,
+        kernel_sizes: List[int] = [3, 5, 7, 9],
+        distance_encoding: Optional[str] = "dot",
+        pre_layer_norm: bool = False,
+        dropout: float = 0.1,
+        temperature: float = 1.0,
+        num_instances_per_series: int = 100,
+        time_features: Optional[List[TimeFeature]] = None,
+        use_feat_dynamic_real: bool = True,
```

Comment:
Does the model support only historical dynamic features as well that you would want to add `use_past_feat_dynamic_real`?

Reply:
No, this model support `feat_dynamic_real` like most implemented models such as DeepAR